### PR TITLE
Add CGBase.h and include CFBase.h to get basic datatypes. Fix #458.

### LINKED ISSUE #458: 'CoreGraphics/CGBase.h' file not found

### PR DESCRIPTION
**Pull request by rajsesh-msft**
Adding CGBase.h.

### CODE REVIEW

**Comment by rajsesh-msft** [comment]
@bbowman is added to the review. #Closed

**Comment by rajsesh-msft** [comment]
@mnithish is added to the review. #Closed

File: include/CoreGraphics/CGBase.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#import <CoreFoundation/CFBase.h>
```
Comment (bbowman) [inline include/CoreGraphics/CGBase.h]: >CFBase.h [](start = 24, length = 8)

Should things in CFBase instead be moved here? Take a look at the comment:

// WINOBJC: Start. Include old WinObjC additions to CFBase.h. Some might not really belong here but are still needed by other dependencies. 

Essientially everything there appears to be CG related ...

**Comment by bbowman** [comment]
:clock1:

**Comment by msft-Jeyaram** [comment]
> // Copyright (c) 2016 Microsoft Corporation. All rights reserved.

should we correct this while at it?

---
Refers to: include/CoreGraphics/CoreGraphics.h:3 in 75b16f4. [](commit_id = 75b16f4a0ab77f01ed4685789f4b31b0bab48f03, deletion_comment = False)

File: include/CoreGraphics/CGBase.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
```
Comment (mnithish) [inline include/CoreGraphics/CGBase.h]: >Copyright (c) 2016 Microsoft Corporation. All rights reserved. [](start = 3, length = 62)

New copyright header?

File: include/CoreGraphics/CGBase.h (hunk 1 above)
Comment (mnithish) [inline include/CoreGraphics/CGBase.h]: +1

---
In reply to: [71259230](https://github.com/Microsoft/WinObjC/pull/664#discussion_r71259230) [](ancestors = 71259230)

**Comment by mnithish** [comment]
:clock1:

**Comment by rajsesh-msft** [comment]
There are some cyclic dependencies between the modules, not surprising due to our reverse dependencies. Moving the CGGraphics definitions strangely surfaces this, haven't had time to investigate.

**Comment by rajsesh-msft** [comment]
Ping @bbowman @mnithish. 

---
In reply to: [236640850](https://github.com/Microsoft/WinObjC/pull/664#issuecomment-236640850) [](ancestors = 236640850)

File: include/CoreGraphics/CGBase.h [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#import <CoreFoundation/CoreFoundation.h>
+
+#ifdef __LP64__
```
Comment (mnithish) [inline include/CoreGraphics/CGBase.h]: >__LP64__ [](start = 7, length = 8)

Do we need this code?

File: include/CoreGraphics/CGColorSpace.h [hunk 4]
```diff
@@ -17,14 +17,16 @@
...
 #import <CoreGraphics/CGDataProvider.h>
 
 COREGRAPHICS_EXPORT void CGColorSpaceRelease(CGColorSpaceRef space);
 COREGRAPHICS_EXPORT CGColorSpaceRef CGColorSpaceRetain(CGColorSpaceRef space);
 COREGRAPHICS_EXPORT const CFStringRef kCGColorSpaceGenericGray;
 COREGRAPHICS_EXPORT const CFStringRef kCGColorSpaceGenericRGB;
 COREGRAPHICS_EXPORT const CFStringRef kCGColorSpaceGenericCMYK;
-;COREGRAPHICS_EXPORT const CFStringRef kCGColorSpaceGenericRGBLinear;
+;
+COREGRAPHICS_EXPORT const CFStringRef kCGColorSpaceGenericRGBLinear;
```
Comment (mnithish) [inline include/CoreGraphics/CGColorSpace.h]: Formatter seems to have done something weird here

Comment (rajsesh-msft) [inline include/CoreGraphics/CGColorSpace.h]: It seems to have broken a line where there was an extra ;

---
In reply to: [75770228](https://github.com/Microsoft/WinObjC/pull/664#discussion_r75770228) [](ancestors = 75770228)

File: include/CoreGraphics/CGBase.h (hunk 3 above)
Comment (rajsesh-msft) [inline include/CoreGraphics/CGBase.h]: Leaving this in with the hope that one day we shall be 64-bit compliant.

---
In reply to: [75770096](https://github.com/Microsoft/WinObjC/pull/664#discussion_r75770096) [](ancestors = 75770096)

File: Frameworks/CoreFoundation/Base.subproj/CFBase.h [hunk 5]
```diff
@@ -680,147 +680,6 @@ CFTypeRef CFMakeCollectable(CFTypeRef cf) CF_AUTOMATED_REFCOUNT_UNAVAILABLE;
...
-#ifdef __LP64__
-typedef double CGFloat;
-#define CGFLOAT_MIN DBL_MIN
-#define CGFLOAT_MAX DBL_MAX
-#else
-typedef float CGFloat;
-#define CGFLOAT_MIN FLT_MIN
-#define CGFLOAT_MAX FLT_MAX
-#endif
-
```
Comment (bbowman) [inline Frameworks/CoreFoundation/Base.subproj/CFBase.h]: this removal is beautiful.

**Pull request merged**